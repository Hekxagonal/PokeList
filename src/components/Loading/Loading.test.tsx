import { screen, render } from '@testing-library/react'
import renderTheme from '../../global/theme/render-theme';

import Component from './'

describe('<Loading />', () => {
    const defaultValues = {
        loadingState: {
            isError: false,
            error: undefined
        }
    }
    it('should render', () => {
        renderTheme(<Component {...defaultValues}/>)

        const pokeball = screen.getByTestId('pokeball')
        expect(pokeball).toBeInTheDocument()
    })
})