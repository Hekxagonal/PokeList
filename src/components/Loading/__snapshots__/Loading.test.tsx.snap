// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  -webkit-animation: fTnmzN 2s ease-in-out infinite;
  animation: fTnmzN 2s ease-in-out infinite;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  position: absolute;
  top: 35.5%;
  left: 39.5%;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  border: 3px solid rgba(0,0,0,1);
  border-top: 3px solid rgba(168,0,0,1);
}

.c3::after {
  position: absolute;
  width: 20px;
  height: 4px;
  content: '';
  background: rgba(0,0,0,1);
  margin: 3px 0px 0px 11px;
}

.c3::before {
  position: absolute;
  width: 20px;
  height: 4px;
  content: '';
  background: rgba(0,0,0,1);
  margin: 3px 0px 0px -22px;
}

.c2 {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: 3px solid rgba(0,0,0,1);
  border-top: 3px solid rgba(168,0,0,1);
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
          data-testid="pokeBall"
        >
          <div
            class="c2"
            data-testid="ballBorder"
          >
            <div
              class="c3"
              data-testid="ballButton"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c1 {
  -webkit-animation: fTnmzN 2s ease-in-out infinite;
  animation: fTnmzN 2s ease-in-out infinite;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  position: absolute;
  top: 35.5%;
  left: 39.5%;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  border: 3px solid rgba(0,0,0,1);
  border-top: 3px solid rgba(168,0,0,1);
}

.c3::after {
  position: absolute;
  width: 20px;
  height: 4px;
  content: '';
  background: rgba(0,0,0,1);
  margin: 3px 0px 0px 11px;
}

.c3::before {
  position: absolute;
  width: 20px;
  height: 4px;
  content: '';
  background: rgba(0,0,0,1);
  margin: 3px 0px 0px -22px;
}

.c2 {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: 3px solid rgba(0,0,0,1);
  border-top: 3px solid rgba(168,0,0,1);
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
        data-testid="pokeBall"
      >
        <div
          class="c2"
          data-testid="ballBorder"
        >
          <div
            class="c3"
            data-testid="ballButton"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
